MekHQ/src/mekhq/campaign/mission/AtBDynamicScenarioFactory.java

```diff
@@ -2366,9 +2378,21 @@ private static void loadBombs(Entity entity, int[] validBombChoices, int year) {
         }
         
         // pick out the index in the BombType array
-        int bombIndex = actualValidBombChoices.get(Compute.randomInt(actualValidBombChoices.size()));
+        int randomBombChoiceIndex = Compute.randomInt(actualValidBombChoices.size());
+        int bombIndex = actualValidBombChoices.get(randomBombChoiceIndex);
+        int weightModifier = 0;
+        
+        // hack: we only really need one "tag", so add it then pack on some more bombs
+        if (bombIndex == BombType.B_TAG) {
+            weightModifier = 5;
+            bombChoices[bombIndex] = 1;
+            actualValidBombChoices.remove(randomBombChoiceIndex);
+            bombIndex = Utilities.getRandomItem(actualValidBombChoices);
+        }
+        
         // # of bombs is the unit's weight / (bomb cost * 5)
-        int numBombs = (int) (entity.getWeight() / (BombType.getBombCost(bombIndex) * 5));
+        int numBombs = (int) ((entity.getWeight() - weightModifier) / 
+                (BombType.getBombCost(bombIndex) * 5));
```

Comment:
Should it be 5.0, to ensure it isn't int / int division? Also, rounding?

Reply:
All the numbers involved in the operation are integers. Rounding down (the default behavior) is correct because otherwise it would result in putting more bombs in than the unit has capacity.

Reply:
To prevent misreads and IDE / CI issues, can I suggest this be written to use the double and a specified round?